python_modules/libraries/dagster-dg/dagster_dg/cli/component.py

```diff
@@ -259,10 +272,76 @@ def component_check_command(
     **global_options: object,
 ) -> None:
     """Check component files against their schemas, showing validation errors."""
+    resolved_paths = [Path(path).absolute() for path in paths]
+    top_level_component_validator = Draft202012Validator(schema=COMPONENT_FILE_SCHEMA)
```

Comment:
do we need to use a specific Validator class like this instead of just `jsonschema.validate()`? This class name seems kinda unfriendly

Reply:
`validate` only raises an exception with the first error; we need a validator in order to access `iter_errors`

Reply:
wild